### DISCUSSION TITLE
splitting multiframe dicom files on C-Store SCU

### DISCUSSION THREAD

**Discussion by mhmmd7777** [fo-dicom/fo-dicom#1904] [Q&A]
How to split multiframe dicom files into separate file when receiving files from modality devices like MR images? Is there option in FellowOakDicom.Network.DicomService to do this?

**Comment by gofal** [comment]
No, there is not. It is not a trivial task, because when you "split up" DICOM-files, then you actually are creating new data out of an existing one. I assume, you receive EnhancedMR data but would prefer to get MRImageStorage data.

If you want to split up the data, then this means you want to generate many MRImageStorage files out of one EnhancedMR image file. You have to create a new SeriesInstanceUID with many new SOPInstanceUIDS, you have to copy some tags of examination parameter in the EnhancedMR to other tags in the MRImageStorage. Depending on the modality, that sends you the EnhancedMR, you will have to map them. You have to read the DICOM conformance statement of the vendor to see which tags are written where. So there is no "general way" that in fo-dicom it can be done automatically.

But I wonder, why should you like to mutate an EnhancedMR to MRImageStorage files? EnhancedMR is perfectly supported in fo-dicom, so what are the problems you are facing with EnhancedMR? There is surely another solution than duplicating the data into another format.

**Comment by mhmmd7777** [comment]
Thank you very much for your reply. MR Images are just examples. I have other multi-frame images like MG, XA, and US images which have more than 50 frames (file size more than 300 MB). Therefore, users have to wait for a large file to receive and review the few frames they want, not all. I want to split them up to make the loading time much faster. Is there any solution for this issue can you offer me? 
Appreciate it.

**Comment by mrbean-bremen** [comment]
This is how the C-STORE protocol works - you will always get the whole dataset. Splitting it afterwards would not make much sense either, I guess.

**Comment by gofal** [comment]
Agree, with C-Store protocol splitting up the file would not speed up anything.
When it comes to access single frames, then fo-dicom should behave in the way that only the requested frame is loaded from disk. If you notice that fo-dicom tries to load the whole file even if you only request one frame, then plase report it and we try to fix that.
